src/hotspot/share/code/codeCache.cpp

```diff
@@ -1772,17 +1783,23 @@ void CodeCache::print_summary(outputStream* st, bool detailed) {
   }
 
   if (detailed) {
-    st->print_cr(" total_blobs=" UINT32_FORMAT " nmethods=" UINT32_FORMAT
-                       " adapters=" UINT32_FORMAT,
+    if (SegmentedCodeCache) {
+      st->print("CodeCache:");
+      st->print_cr(" size=" JULONG_FORMAT "Kb, used=" JULONG_FORMAT
+                  "Kb, max_used=" JULONG_FORMAT "Kb, free=" JULONG_FORMAT "Kb",
+                  total_size, total_used, total_max_used, total_free);
+    }
+    st->print_cr(" total_blobs=" UINT32_FORMAT ", nmethods=" UINT32_FORMAT
+                       ", adapters=" UINT32_FORMAT,
                        blob_count(), nmethod_count(), adapter_count());
-    st->print_cr(" compilation: %s", CompileBroker::should_compile_new_jobs() ?
-                 "enabled" : Arguments::mode() == Arguments::_int ?
-                 "disabled (interpreter mode)" :
-                 "disabled (not enough contiguous free space left)");
-    st->print_cr("              stopped_count=%d, restarted_count=%d",
+    st->print_cr(" stopped_count=%d, restarted_count=%d, full_count=%d",
                  CompileBroker::get_total_compiler_stopped_count(),
-                 CompileBroker::get_total_compiler_restarted_count());
-    st->print_cr(" full_count=%d", full_count);
+                 CompileBroker::get_total_compiler_restarted_count(),
+                 full_count);
+    st->print_cr(" compilation=%s", CompileBroker::should_compile_new_jobs() ?
+              "enabled" : Arguments::mode() == Arguments::_int ?
+              "disabled (interpreter mode)" :
+              "disabled (not enough contiguous free space left)");
```

Comment:
Why did you change the order of the `compilation=` and the `stopped_count=` output?